### PR TITLE
Add a dev container supporting docs and frontend

### LINKED ISSUE #1022: Configure Codespaces as a quick way to contribute to Openverse
## Problem

Since the project needs a lot of dependencies like Docker, Python, Node and `just` (some of which can be hard to set up), configuring everything can be time-consuming and difficult.

## Description

Codespaces can provide a nice way to work around this issue by providing pre-built ready-to-go environments. We can [configure our codespace](https://docs.github.com/en/codespaces/setting-up-your-project-for-codespaces/adding-a-dev-container-configuration/introduction-to-dev-containers) set up our environment, define packages and languages to pre-install and make the codespace self-sufficient, at least for frontend, automations and documentation stacks.

Also, for minor contributions like documentation changes or making tiny changes in an existing PR, codespaces are easier compared to setting things up locally.

## Additional context

Just for clarity we [will not be charged](https://docs.github.com/en/billing/managing-billing-for-github-codespaces/about-billing-for-github-codespaces) for contributors who spin up codespaces from our repo. Everyone gets a certain amount of free usage, which should be pretty good to get them familiar with the code and how to run it locally.

### PR DESCRIPTION
**Pull request by dhruvkb**
## Fixes

Fixes #1022 by @dhruvkb 

## Description

This PR sets up a dev container that can be used locally (which is awesome) or as a GitHub codespace (not as awesome but still usable).

This container contains everything you need to be immediately productive in two stacks (for now, others maybe later), the documentation and the frontend.

## Testing Instructions

- Navigate to `WordPress/openverse`.
- Click on the "<> Code" button.
- Switch to the Codespaces tab.
- Click on the "..." dropdown button.
- Choose "New with options".
- Change branch to this one `dev_container`. This should automatically apply the "Openverse" dev container configuration.
- Customise any other options to your liking.
- Create the codespace.

The codespace can take quite some time to build and initialise. Grab a coffee or snack in the meantime. Or go for a walk. Whatever floats your boat. During this step the container will set up all pnpm dependencies, install Python projects that run outside Docker and set up linting with pre-commit.

### Documentation

- Run `just documentation/live` in the attached terminal.
- Switch to the ports tab (adjacent to the terminal).
- Click on the "🌐" icon next to the 50230 port. That should open the forwarded port in a new tab. 
- Make some changes to the documentation. Your changes should reflect in the preview in real time.

Screenshot 2023-05-07 at 6 19 07 PM

### Frontend

- Run `just frontend/run dev` in the attached terminal.
- Switch to the ports tab (adjacent to the terminal).
- Click on the "🌐" icon next to the 8443 port. That should open the forwarded port in a new tab. 
- Make some changes to the frontend. Your changes should reflect in the preview in real time.

Screenshot 2023-05-07 at 6 44 55 PM

## Caveats

The dev container is frustratingly slow and pretty limited in what it can do, considering it's all running _inside_ a Docker container. Additionally it's currently limited to the documentation and the frontend (only if you opt for a higher RAM codespace). Dockerised applications like the API will run with Docker-in-Docker and as I understand, will be even slower 😢.

But for a lot of contributors who struggle with setting up an environment complete will all the pre-requisites, this is amazing. It has all you need to be productive and make a quick PR to two stacks!

## Checklist

- [x] My pull request has a descriptive title (not a vague title like`Update index.md`).
- [x] My pull request targets the _default_ branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.
- [ ] I ran the DAG documentation generator (if applicable).

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Review by sarayourfriend — APPROVED** [review]
LGTM! It truly is dog slow using GitHub codespaces :cold_sweat:. But glad that it would help frontend contributions be a bit easier, especially as we're on a node version that is reaching EOL.

File: .devcontainer/devcontainer.json
Comment (krysal) [inline .devcontainer/devcontainer.json:1]: This folder needs to be added to the CODEOWNERS file.

**Review by krysal — APPROVED** [review]
Great DX improvement. LGTM.

**Review by krysal — APPROVED** [review]
Great DX improvement. LGTM.

File: .devcontainer/devcontainer.json
Comment (dhruvkb) [reply .devcontainer/devcontainer.json:1]: Thanks for the heads-up, added in 0d1e61f.

**Pull request merged**